psi4/src/psi4/libfunctional/superfunctional.cc

```diff
@@ -801,84 +807,124 @@ std::map<std::string, SharedVector> SuperFunctional::compute_vv10_cache(
 
     // Kernel between left (*this) and right (vv10_cache) grids
 
     // Compute the vv10 cache in place
-    double l_thresh = 1.e-12;
-    size_t l_npoints = block->npoints();
+    const double l_thresh = 1.e-12;
+    const size_t l_npoints = block->npoints();
     compute_vv10_cache(vals, block, l_thresh, block->npoints(), true);
 
     // Grab values to update
     double vv10_e = 0.0;
     double* v_rho = values_["V_RHO_A"]->pointer();
     double* v_gamma = values_["V_GAMMA_AA"]->pointer();
+    double* x_grid = vv_values_["GRID_WX"]->pointer();
+    double* y_grid = vv_values_["GRID_WY"]->pointer();
+    double* z_grid = vv_values_["GRID_WZ"]->pointer();
+    std::fill(x_grid, x_grid+l_npoints, 0.0);
+    std::fill(y_grid, y_grid+l_npoints, 0.0);
+    std::fill(z_grid, z_grid+l_npoints, 0.0);
 
     // Constants
     const double vv10_beta = vv10_beta_;
 
     // Get left points
-    double* l_x = block->x();
-    double* l_y = block->y();
-    double* l_z = block->z();
-    double* l_w = block->w();
-    double* l_rho = vals.find("RHO_A")->second->pointer();
-    double* l_gamma = vals.find("GAMMA_AA")->second->pointer();
-    double* l_W0 = vv_values_["W0"]->pointer();
-    double* l_kappa = vv_values_["KAPPA"]->pointer();
+    const double* l_x = block->x();
+    const double* l_y = block->y();
+    const double* l_z = block->z();
+    const double* l_w = block->w();
+    const double* l_rho = vals.find("RHO_A")->second->pointer();
+    const double* l_gamma = vals.find("GAMMA_AA")->second->pointer();
+    const double* l_W0 = vv_values_["W0"]->pointer();
+    const double* l_kappa = vv_values_["KAPPA"]->pointer();
 
     for (size_t i = 0; i < l_npoints; i++){
 
         // Add Phi agnostic quantities
         vv10_e += l_w[i] * l_rho[i] * vv10_beta;
-        v_rho[i] += vv10_beta;
+        v_rho[i] = vv10_beta;
+        v_gamma[i] = 0.0;
 
         if (l_rho[i] < l_thresh) continue;
 
         // Compute interior kernel
         double phi = 0.0;
         double U = 0.0;
         double W = 0.0;
+        double xc = 0.0;
+        double yc = 0.0;
+        double zc = 0.0;
         for (auto r_block : vv10_cache){
 
             // Get right points
-            double* r_x = r_block["X"]->pointer();
-            double* r_y = r_block["Y"]->pointer();
-            double* r_z = r_block["Z"]->pointer();
-            double* r_w = r_block["W"]->pointer();
-            double* r_rho = r_block["RHO"]->pointer();
-            double* r_W0 = r_block["W0"]->pointer();
-            double* r_kappa = r_block["KAPPA"]->pointer();
+            const double* r_x = r_block["X"]->pointer();
+            const double* r_y = r_block["Y"]->pointer();
+            const double* r_z = r_block["Z"]->pointer();
+            const double* r_w = r_block["W"]->pointer();
+            const double* r_rho = r_block["RHO"]->pointer();
+            const double* r_W0 = r_block["W0"]->pointer();
+            const double* r_kappa = r_block["KAPPA"]->pointer();
```

Comment:
These pointers may have to be marked `RESTRICT` to get the SIMD benefits below

Reply:
oops, they're const so it should be fine

Reply:
The opt report shows that this is properly vectorized. Still getting this in VTune would be worthwhile checking- it looks like an easy vectorization, but isn't always the case.